### DISCUSSION TITLE
Transactional

### DISCUSSION THREAD

**Discussion by starshb** [starshb/pharmacy_Find#9] [Q&A] [answered]
Transactional

**Comment by tjdud22** [comment]
트랜잭션(Transaction)은 데이터베이스의 상태를 변환시키는 하나의 논리적 기능을 수행하기 위한 작업의 단위 또는 한꺼번에 모두 수행되어야 할 일련의 연산들을 의미한다.

**Comment by tjdud22** [comment, marked as answer]
DB에서 상태변화의 한 주기를 의미한다.
바로 이전 커밋 이후의 작업부터, 이런저런 작업 후 최종적으로 커밋을 날리기까지의 주기이다.
@Transactional 어노테이션은, 해당 어노테이션이 적용되는 메소드를 하나의 트랜잭션으로 묶어주는 역할을 한다.

**Comment by tjdud22** [comment]
개별 메소드나 클래스에 붙을 수 있는 트랜잭션 특성.
class레벨의 어노테이션은 선언된 클래스와 그의 서브클래스 내부에 있는 모든 메소드에 적용된다.
조상 클래스로 올라가 적용되지는 않는다고 한다!
